### PR TITLE
Updating to support subnet. Fixing development server network issue.

### PR DESCRIPTION
**Pull request by waspper**

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -1,3 +1,4 @@
+.directory
```
[the remark this answers is not in the document]
Comment (waspper) [reply .gitignore]: It's not a folder. That file is created under Linux-like OS (at least, using KDE and Dolphin) to store information like this (example):
[Dolphin]
Timestamp=2018,1,19,15,8,42
Version=3

[Settings]
HiddenFilesShown=true

**Comment by waspper** [comment]
-- PR was cleaned-up.

File: .gitignore (hunk 1 above)
[the remark this answers is not in the document]
Comment (andypost) [reply .gitignore]: It should not be added twice

File: Makefile [hunk 2]
```diff
@@ -122,16 +116,17 @@ clean: info down
 	@for i in $(DIRS); do if [ -d "web/$$i" ]; then echo "Removing web/$$i..."; docker run --rm -v $(shell pwd):/mnt $(IMAGE_PHP) sh -c "rm -rf /mnt/web/$$i"; fi; done
 
 net:
-ifeq ($(shell docker network ls -q -f Name=$(COMPOSE_NET_NAME)),)
+ifeq ($(strip $(shell docker network ls | grep $(COMPOSE_NET_NAME))),)
+ifneq ($(COMPOSE_SUBNET_RANGE),)
```
Comment (andypost) [inline Makefile]: no reason in extra var

the logic here is simple
1) if net exists (`docker network ls` returns it) & no `IPRANGE` in `.env` then "read subnet from existing net and put to .IPRANGE"
2) if there's no net & no range defined in .env - call `docker network create $(COMPOSE_NET_NAME)` to make docker allocate new net for use & then  "read subnet from existing net and put to .IPRANGE"

File: Makefile [hunk 3]
```diff
@@ -122,16 +116,17 @@ clean: info down
...
+	docker network create --subnet $(COMPOSE_SUBNET_RANGE) $(COMPOSE_NET_NAME)
```
Comment (andypost) [inline Makefile]: do not pass `--subnet` here - docker should give you new range when net will be created, you just need populate IPRANGE with it in .env file

Comment (tkyryliuk) [reply Makefile]: we should not run `docker network create` when we already have IPRANGE

**Comment by waspper** [comment]
No longer needed.

**Pull request closed without merging**